File: python/maplestory_openapi/api/dto/character/character_skill.py
from datetime import datetime
from pydantic import BaseModel


class CharacterSkillInfo(BaseModel):
    """스킬 정보

    skill_name(str): 스킬 명
    skill_description(str): 스킬 설명
    skill_level(int): 스킬 레벨
    skill_effect(str): 스킬 레벨 별 효과 설명
    skill_effect(str): 다음 스킬 레벨 효과 설명
    skill_icon(str): 스킬 아이콘
    """
    skill_name: str
    skill_description: str
    skill_level: int
    skill_effect: str | None
    skill_effect_next: str | None
    skill_icon: str


class CharacterSkill(BaseModel):
    """캐릭터 스킬 정보

    date(datetime): 조회 기준일 (KST, 일 단위 데이터로 시, 분은 일괄 0으로 표기)
    character_class(str): 캐릭터 직업
    character_skill_grade(str): 스킬 전직 차수
    character_skill(list[CharacterSkillInfo]): 스킬 정보
    """
    date: datetime | None
    character_class: str
    character_skill_grade: str
    character_skill: list[CharacterSkillInfo]

File: js/src/maplestory/api/kms/dto/character/characterSkill.ts
import * as base from '../../../common/dto/character/characterSkill';
import {
  CharacterSkillBody,
  CharacterSkillInfoBody,
} from '../../response/character/characterSkillBody';

/**
 * 캐릭터 스킬 정보
 */
export class CharacterSkillDto extends base.CharacterSkillDto {
  /**
   * 조회 기준일 (KST, 일 단위 데이터로 시, 분은 일괄 0으로 표기)
   */
  public override date: Date | null;

  /**
   * 캐릭터 직업
   */
  public override characterClass: string | null;

  /**
   * 스킬 전직 차수
   */
  public override characterSkillGrade: string | null;

  /**
   * 스킬 정보
   */
  public override characterSkill: CharacterSkillInfoDto[];

  constructor(obj: CharacterSkillBody) {
    super();

    const { date, character_class, character_skill_grade, character_skill } =
      obj;

    this.date = date ? new Date(date) : null;
    this.characterClass = character_class;
    this.characterSkillGrade = character_skill_grade;
    this.characterSkill = character_skill.map(
      (skill) => new CharacterSkillInfoDto(skill),
    );
  }
}

/**
 * 캐릭터 스킬 상세 정보
 */
export class CharacterSkillInfoDto extends base.CharacterSkillInfoDto {
  /**
   * 스킬 명
   */
  public override skillName: string;

  /**
   * 스킬 설명
   */
  public override skillDescription: string;

  /**
   * 스킬 레벨
   */
  public override skillLevel: number;

  /**
   * 스킬 레벨 별 효과 설명
   */
  public override skillEffect: string | null;

  /**
   * 다음 스킬 레벨 효과 설명
   */
  public override skillEffectNext: string | null;

  /**
   * 스킬 아이콘
   */
  public override skillIcon: string;

  constructor(obj: CharacterSkillInfoBody) {
    super();

    const {
      skill_name,
      skill_description,
      skill_level,
      skill_effect,
      skill_effect_next,
      skill_icon,
    } = obj;

    this.skillName = skill_name;
    this.skillDescription = skill_description;
    this.skillLevel = skill_level;
    this.skillEffect = skill_effect;
    this.skillEffectNext = skill_effect_next;
    this.skillIcon = skill_icon;
  }
}
